### DISCUSSION TITLE
TypeError: can't read property of `args` while $extend to $allOperations

### DISCUSSION THREAD

**Discussion by meotimdihia** [prisma/prisma#22652] [Q&A] [answered]
### Bug description

Property 'data' does not exist on type 'miesFindFirstArgs<InternalArgs & DefaultArgs> | miesCreateArgs<InternalArgs & DefaultArgs> | ... 12 more ... | miesGroupByArgs<...>'.
  Property 'data' does not exist on type 'miesFindFirstArgs<InternalArgs & DefaultArgs>'.
### How to reproduce

```ts
  prisma.$extends({
    query: {
      mies: {

        async $allOperations({ model, operation,  query , args}) {
          if (
            model == "mies" &&
            ["create", "upsert"].indexOf(operation) != -1
          ) {
            if (args.data) {
        
         
```
![image](https://github.com/prisma/prisma/assets/300961/3f0de3c6-c1f2-49ca-b67d-b5402530ef48)

### Prisma information

Not necessary. It is easy to reproduce.

### Environment & setup

- OS: MacOS
- Database: PosgreSQL
- Node.js version: 20

### Prisma Version

```
prisma                  : 5.8.0
@prisma/client          : 5.8.0
```

**Comment by SevInf** [comment, marked as answer]
` ["create", "upsert"].indexOf(operation) != -1` is not something TS can understand and narrow down the types of `args` down. Try replacing this line with `operation === 'create' || operation === 'upsert'`.

**Reply by meotimdihia** [reply]
Thanks, it works for me 👍
